apps/aecore/lib/aecore/structures/spend_tx.ex

```diff
@@ -2,38 +2,108 @@ defmodule Aecore.Structures.SpendTx do
   @moduledoc """
   Aecore structure of a transaction data.
   """
+
+  @behaviour Aecore.Structures.Transaction
+
   alias Aeutil.Serialization
-  alias Aecore.Structures.SpendTx
 
-  @type t :: %SpendTx{
-    from_acc: binary(),
+  @typedoc "Arbitrary structure data of a transaction"
+  @type payload ::map()
+
+  @typedoc "Reason for the error"
+  @type reason :: String.t()
+
+  @typedoc "Structure of the Spend Transaction type"
+  @type t :: %__MODULE__{
     to_acc: binary(),
     value: non_neg_integer(),
-    nonce: non_neg_integer(),
-    fee: non_neg_integer(),
     lock_time_block: non_neg_integer()
   }
 
   @doc """
   Definition of Aecore SpendTx structure
 
   ## Parameters
-  - nonce: A random integer generated on initialisation of a transaction.Must be unique
-  - from_acc: From account is the public address of one account originating the transaction
   - to_acc: To account is the public address of the account receiving the transaction
   - value: The amount of a transaction
+  - lock_time_block: To which block the funds will become available
   """
-  defstruct [:from_acc, :to_acc, :value, :nonce, :fee, :lock_time_block]
+  defstruct [:to_acc, :value, :lock_time_block]
   use ExConstructor
 
-  @spec create(binary(), binary(), non_neg_integer(), non_neg_integer(), non_neg_integer(), non_neg_integer()) :: {:ok, SpendTx.t()}
-  def create(from_acc, to_acc, value, nonce, fee, lock_time_block \\ 0) do
-    {:ok, %SpendTx{from_acc: from_acc,
-                  to_acc: to_acc,
-                  value: value,
-                  nonce: nonce,
-                  fee: fee,
-                  lock_time_block: lock_time_block}}
+  @spec init(payload()) :: SpendTx.t()
+  def init(%{to_acc: to_acc, value: value, lock_time_block: lock} = payload) do
+    %__MODULE__{to_acc: to_acc,
+                value: value,
+                lock_time_block: lock}
+  end
+
+  @spec is_valid(SpendTx.t()) :: :ok | {:error, reason()}
+  def is_valid(%__MODULE__{value: value}) do
+    if value >= 0 do
+      :ok
+    else
+      {:error, "Value not enough"}
```

Comment:
I think the error message can be changed to something more meaningful like "Value cannot be a negative number"